DuckDuckGo/UnifiedFeedbackForm/UnifiedFeedbackSender.swift

```diff
@@ -0,0 +1,118 @@
+//
+//  UnifiedFeedbackSender.swift
+//
+//  Copyright © 2023 DuckDuckGo. All rights reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+import Foundation
+import PixelKit
+
+protocol UnifiedFeedbackSender {
+    func sendFeatureRequestPixel(description: String, source: String) async throws
+    func sendGeneralFeedbackPixel(description: String, source: String) async throws
+    func sendReportIssuePixel<T: UnifiedFeedbackMetadata>(source: String, category: String, subcategory: String, description: String, metadata: T?) async throws
+
+    func sendFormShowPixel()
+    func sendSubmitScreenShowPixel(source: String, reportType: String, category: String, subcategory: String)
+    func sendSubmitScreenFAQClickPixel(source: String, reportType: String, category: String, subcategory: String)
+}
+
+extension UnifiedFeedbackSender {
+    func sendStandardPixel(_ pixel: PixelKitEventV2) async throws {
+        try await withCheckedThrowingContinuation { (continuation: CheckedContinuation<Void, Error>) in
+            PixelKit.fire(pixel, frequency: .standard) { _, error in
+                if let error {
+                    continuation.resume(throwing: error)
+                } else {
+                    continuation.resume()
+                }
+            }
+        }
+    }
+}
+
+protocol StringRepresentable: RawRepresentable {
+    static var `default`: Self { get }
+}
+
+extension StringRepresentable where RawValue == String {
+    static func from(_ text: String) -> String {
+        (Self(rawValue: text) ?? .default).rawValue
+    }
+}
```

Comment:
Interesting - not sure how I feel about this yet, but it's neat 😄

Reply:
Ideally I'd pass the strong types around, but making it work with the `UnifiedFeedbackSender` is tricky so I settled for this :D